src/main/java/io/appium/java_client/android/AndroidMobileOptions.java

```diff
@@ -0,0 +1,1733 @@
+     */
+    public AndroidMobileOptions setAndroidCoverageEndIntent(String coverage) {
+        return amend(AndroidMobileCapabilityType.ANDROID_COVERAGE_END_INTENT, coverage);
+    }
+
+    /**
+     * Get the broadcast action which is used to dump coverage into file system.
+     *
+     * @return String of the broadcast action which is used to dump coverage into file system.
+     * @see AndroidMobileCapabilityType#ANDROID_COVERAGE_END_INTENT
+     */
+    public String getAndroidCoverageEndIntent() {
+        return (String) getCapability(AndroidMobileCapabilityType.ANDROID_COVERAGE_END_INTENT);
+    }
+
+    /**
+     * Set the timeout to wait for a device to become ready after booting.
+     *
+     * @param duration is the number of seconds to wait for a device to become ready after booting.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ANDROID_DEVICE_READY_TIMEOUT
+     */
+    public AndroidMobileOptions setAndroidDeviceReadyTimeout(Duration duration) {
+        return amend(AndroidMobileCapabilityType.ANDROID_DEVICE_READY_TIMEOUT, duration.getSeconds());
+    }
+
+    /**
+     * Get the timeout to wait for a device to become ready after booting.
+     *
+     * @return Duration to wait for a device to become ready after booting.
+     * @see AndroidMobileCapabilityType#ANDROID_DEVICE_READY_TIMEOUT
+     */
+    public Duration getAndroidDeviceReadyTimeout() {
+        Object duration = getCapability(AndroidMobileCapabilityType.ANDROID_DEVICE_READY_TIMEOUT);
+        if (duration.getClass().isAssignableFrom(Long.class)) {
+            return Duration.ofSeconds((Long) duration);
+        }
+        Integer integer = (Integer) duration;
+        Long value = Long.valueOf(integer);
+        return Duration.ofSeconds(value);
+    }
+
+    /**
+     * Set the devtools socket name.
+     *
+     * @param activity is the devtools socket name.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ANDROID_DEVICE_SOCKET
+     */
+    public AndroidMobileOptions setAndroidDeviceSocket(String activity) {
+        return amend(AndroidMobileCapabilityType.ANDROID_DEVICE_SOCKET, activity);
+    }
+
+    /**
+     * Get the devtools socket name.
+     *
+     * @return String of the devtools socket name.
+     * @see AndroidMobileCapabilityType#ANDROID_DEVICE_SOCKET
+     */
+    public String getAndroidDeviceSocket() {
+        return (String) getCapability(AndroidMobileCapabilityType.ANDROID_DEVICE_SOCKET);
+    }
+
+    /**
+     * Set the name of the directory on the device in which the apk will be pushed before install.
+     *
+     * @param path is the directory on the device in which the apk will be pushed before install.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ANDROID_INSTALL_PATH
+     */
+    public AndroidMobileOptions setAndroidInstallPath(String path) {
```

Comment:
this cap is not used anywhere in the code: https://github.com/appium/appium/search?q=androidInstallPath&unscoped_q=androidInstallPath&type=Code